main/src/ca/uwaterloo/flix/language/CompilationMessageKind.scala

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright 2024 Jakob Schneider Villumsen
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package ca.uwaterloo.flix.language
+
+import ca.uwaterloo.flix.language.CompilationMessageKind.*
+import ca.uwaterloo.flix.language.ast.Ast.SyntacticContext
+
+sealed trait CompilationMessageKind extends Ordered[CompilationMessageKind] {
+  override def toString: String = this match {
+    case DerivationError => "Derivation Error"
+    case EntryPointError => "Entry Point Error"
+    case InstanceError => "Instance Error"
+    case KindError => "Kind Error"
+    case LexerError => "Lexer Error"
+    case NameError => "Name Error"
+    case ParseError(sctx) => s"Parse Error ($sctx)"
+    case PatternMatchError => "Pattern Match Error"
+    case RedundancyError => "Redundancy Error"
+    case ResolutionError => "Resolution Error"
+    case SafetyError => "Safety Error"
+    case StratificationError => "Stratification Error"
+    case TestError => "Test Error"
+    case TypeError => "Type Error"
+    case WeederError => "Syntax Error"
+  }
+
+  override def compare(that: CompilationMessageKind): Int = (this, that) match {
```

Comment:
What is this used for?

Reply:
The order instance as described in the issue / ticket. I can drop the instance though since we are not going to do filtering

Reply:
Ah yes. lets drop it or now.